### ISSUE TITLE
poco thread detach from GUI  bug feedback 

### ISSUE THREAD

**Issue by lion117** [pocoproject/poco#695]
when i tried to apply  poco::thread  into MFC dailog to dea iwth process control , it didnot work which would crash 
 i had tried to use lambda to solve this problem . but stil didnot work  . 
to test whether it was a bug , i had apply  C++ 11 stl  new class thread to solve this problem . it could work .  following is std  solution for GUI  thread 
//.cpp   void CMFClambdaDlg::OnBnClickedButton1() function 
    CProgressCtrl *pprocess = (CProgressCtrl *)GetDlgItem(IDC_PROGRESS1);
    pprocess->SetRange(0, 99);
    pprocess->SetStep(1);
    pprocess->SetPos(0);
    auto func = [=]
    {
        for (int i = 0; i < 100; i++)
        {
            pprocess->SetPos(i);
            Sleep(10);
        }
    };
    std::thread mythead(func);
    mythead.detach();                          // key point for GUI 
////////////////////////////////////////////////////////////////////////////////////////////////////////////
upper code could work  

// .cpp    poco thread 
    CProgressCtrl *pprocess = (CProgressCtrl *)GetDlgItem(IDC_PROGRESS1);
    pprocess->SetRange(0, 99);
    pprocess->SetStep(1);
    pprocess->SetPos(0);
    auto func = [=]
    {
        for (int i = 0; i < 100; i++)
        {
            pprocess->SetPos(i);
            Sleep(10);
        }
    };
    Thread mythread;
    mythread.startFunc(func);
    mythread.join();               // if prohibit this code line , then crashed  
////////////////////////////////////////////////////////////////////////////////////////////////////////////
// resoult :  nothing occure 

as i had notice that normal thread could not operation GUI control  directly  ,  usually  i would have to postmessage  to do this job which is really un-convenient  , is it possible for poco::thread to make something better ?  like std::thread had done ?

**Comment by aleks-f** [comment]
Yes, you can achieve the same effect if you create Poco:Thread on the heap. As you have it now, it is created on the stack and destroyed when the function where you created it completes (unless, of course, you call join() which blocks until thread completes, as you have noticed).
